### PR TITLE
 support audio & audio-text data reading

### PR DESCRIPTION
**Pull request by chenhesen**
1. support audio & audio-text data reading
 2. add wavcaps_to_dj and dj_to_wavcaps tool

### CODE REVIEW

File: tools/multimodal/README_ZH.md [hunk 1]
```diff
@@ -74,3 +75,37 @@ python tools/multimodal/source_format_to_data_juicer_format/llava_to_dj.py --hel
     }
 ]
 ```
+
+#### 类WavCaps格式
+[WavCaps](https://github.com/XinhaoMei/WavCaps#dataset) 数据集由 [FreeSound](https://freesound.org/)，[BBC Sound Effects](https://sound-effects.bbcrewind.co.uk/)，[SoundBible](https://soundbible.com/)，[AudioSet Strongly-labelled Subset](https://research.google.com/audioset/download_strong.html) 四个子数据集组成，每个数据集里都有不同的字段。例如SoundBible里包含了‘description’字段，而该字段在AudioSet里并不存在。为了保证不同子数据集在转换后能够正常合并，在wavcaps_to_dj阶段使用了所有子数据集字段的并集，并在dj_to_wavcaps阶段完整保留了所有字段。
```
Comment (HYLcool) [inline tools/multimodal/README_ZH.md:80]: Notice that this dataset didn't declare its dataset format in detail, so here we can list all fields that will be kept in the end for better usage.

Same in the English version.

Comment (chenhesen) [reply tools/multimodal/README_ZH.md:80]: All fields are listed in the case below

Comment (HYLcool) [reply tools/multimodal/README_ZH.md:80]: It's **better** to list them here in the **Notice** section in the **doc**.

File: tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+#       'caption': 'An airplane is landing.',
+#       'id': '2219',
+#       'duration': 14.1424375,
+#       'audio': 'wav_path',
+#       'download_link': 'http://soundbible.com/grab.php?id=2219&type=wav',
+#       'category': '',
+#       'tags': '' }}
+# {'audios': ['./path/to/audio/2218.flac'],
+#  'text': '<audio>\n'
+#          '[[caption]]: Someone is ringing a bell. <|__dj__eoc|>',
```
Comment (HYLcool) [inline tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py:49]: The string "[[caption]]: " might be unnecessary? LLaVA format needs this "role" part to represent single/multi-turn conversations, but there is no need to add this part for WavCaps, which is actually a captioning dataset.

File: tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+
+@logger.catch
+def main(
+    wavcaps_json_path: str,
+    wavcaps_audio_path: str,
+    target_ds_path: str,
+    target_field: Union[str, List[str]] = 'caption',
+    eoc_special_token: str = SpecialTokens.eoc,
+    audio_special_token: str = '<audio>',
```
Comment (HYLcool) [inline tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py:113]: If there is no explicit special token specified by the original dataset, it's recommended to use the default special token from Data-Juicer, which is SpeicalTokens.audio here.

LLaVA use "<image>" in the original dataset to indicate the position of audios, but WavCaps didn't specify such tokens as far as I know.

File: tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    :param target_ds_path: path to store the converted dataset in Data-Juicer
+        format.
+    :param target_field: the field used to describe audio in the WavCaps-like
+        dataset, which can be one or more of ['caption','title','description'].
+    :param eoc_special_token: the special token for "end of a chunk". It's used
+        to split conversation chunks explicitly. Default: <|__dj__eoc|> (from
+        Data-Juicer).
+    :param audio_special_token: the special token for audios. It's used to
+        locate the audios in the text. In typical WavCaps-like datasets,
+        this token always be "<audio>". You can change it to align with your
```
Comment (HYLcool) [inline tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py:132]: As the above comment says, it seems that WavCaps didn't specify any tokens for audio.

File: tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+                    f'No audios in the sample with id [{audio_name}], '
+                    f'which means this sample is not a multimodal '
+                    f'sample. You\'d better remove this sample '
+                    f'before converting.')
+                continue
+            audio = [all_audio_files[audio_name]]
+            text = audio_special_token + sent_seperator
+            if target_field not in sample.keys():
+                logger.warning(
+                    f'{target_field} does not exist in this sample.')
```
Comment (HYLcool) [inline tools/multimodal/source_format_to_data_juicer_format/wavcaps_to_dj.py:198]: Give the sample id in this log to help users find it.

File: tools/multimodal/data_juicer_format_to_target_format/dj_to_wavcaps.py [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    if target_field not in ['caption', 'description', 'title']:
+        raise ValueError(
+            "target_field must be in '['caption', 'description', 'title']'")
+
+    logger.info('Start to convert.')
+    samples = {'num_captions_per_audio': 1, 'data': []}
+    with jl.open(dj_ds_path, 'r') as reader:
+        for sample in tqdm(reader):
+            if Fields.meta not in sample:
+                logger.warning(f'{Fields.meta} does not exist in this sample.')
```
Comment (HYLcool) [inline tools/multimodal/data_juicer_format_to_target_format/dj_to_wavcaps.py:132]: need to show sample id

File: tools/multimodal/data_juicer_format_to_target_format/dj_to_wavcaps.py [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+                continue
+
+            if target_field not in sample[Fields.meta].keys():
+                logger.warning(
+                    f'{target_field} does not exist in this sample.')
```
Comment (HYLcool) [inline tools/multimodal/data_juicer_format_to_target_format/dj_to_wavcaps.py:137]: need to show sample id

**Review by HYLcool — APPROVED** [review]
LGTM now.

**Pull request merged**